Rename Chinese translation files. Closes #2936.

sledgehammer999 opened the pull request:
@Chocobo1 can you test this?

Test if the language selection from the GUI works. (I forgot to do the webui).
Also test if from a new install the correct locale is selected. eg on a German computer a clean install results in the German locale to be select upon first start.

And try to do a more general testing.
Waiting your info.

PS: After this is merged I'll create the relevant locales on Transifex too.

Chocobo1 commented:
> Language selection from the GUI

It works, but the language selection in options doesn't show correctly, it's blank.

After some tracing, it seems [options_imp.cpp#L1078](https://github.com/qbittorrent/qBittorrent/blob/a2c48646ae201b73242a4fc7f052d351e07084a9/src/gui/options_imp.cpp#L1080) is interpreting data differently than [options_imp.cpp#L285](https://github.com/qbittorrent/qBittorrent/blob/a2c48646ae201b73242a4fc7f052d351e07084a9/src/gui/options_imp.cpp#L285), i.e. `locale.name()` is returning `ISO 639 + ISO 3166` while `QString localeStr` is `ISO 639 + ISO 15924`.
Better read the qt doc for better explanation...

> Test if from a new install the correct locale is selected

The one-time legal notice text falls back to English, something is wrong here...


Could you give a windows build to test? My own builds on windows is affected by #3059, had to patch things out...

Chocobo1 commented on src/lang/qbittorrent_zh_Hans.ts:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!DOCTYPE TS>
-<TS version="2.0" language="zh">
+<TS version="2.0" language="zh-Hans">
```
shouldn't this be `zh_Hans`?

Chocobo1 commented on src/lang/qbittorrent_zh_Hant.ts:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <!DOCTYPE TS>
-<TS version="2.0" language="zh_TW">
+<TS version="2.0" language="zh-Hant">
```
shouldn't this be `zh_Hant`?

sledgehammer999 replied:
I am not sure. I temporarily created the Chinese Traditional Locale on Transifex and downloaded the file. This is what it had...
However, I don't know how I can actually test that qt picks up this as Chinese Traditional and doesn't expect "zh_Hant".

Chocobo1 replied:
Maybe this difference can be ignored if the blank QComboBox issue can be solved.

sledgehammer999 commented:
I updated the code.
Here is a build too: http://builds.shiki.hu/temp/qbittorrent_3.3.0alpha_based_on_a2c48646ae20.7z

Chocobo1 commented:
Tried the build. The QComboBox issue is fixed.

The one-time legal notice text is still in English, i.e. qbt couldn't detect the correct language to use on start up.
The whole interface is in English.
When opening the Option window for the first time, the lang QComboBox shows `中文 (简体)` (zh_Hans), but the correct one should be `中文 (繁體)` (zh_Hant).

sledgehammer999 commented:
I updated the code and I think I have fixed the "one time legal issue" thing.
I also hope it fixed this too:
>When opening the Option window for the first time, the lang QComboBox shows 中文 (简体) (zh_Hans), but the correct one should be 中文 (繁體) (zh_Hant).

I have it to detect if the system locale is Chinese language. If it is try to query the script. If it is anything else than traditional(or empty) it defaults to Simplified. Is that ok? Or should I make Traditional Chinese default?

Here is a build too: http://builds.shiki.hu/temp/qbittorrent_3.3.0alpha_based_on_a2c48646ae20_rev1.7z

Chocobo1 commented:
Update: Now the legal text is in Chinese alright, but it's in zh_Hans, the correct one should be zh_Hant. 

> Or should I make Traditional Chinese default?

Yes, let's try this, i.e.
```
if( script() == QLocale::SimplifiedHanScript)
  = "zh_Hans";
else
  = "zh_Hant";
```

sledgehammer999 commented:
>but it's in zh_Hans, the correct one should be zh_Hant. 

Then QLocale doesn't work perfectly. QSystem::Locale::script() probably returns QLocale::AnyScript.
But let me try one last thing.

sledgehammer999 commented:
>But let me try one last thing.

Forget this.

I made Traditional Chinese as default.
Here is a build: http://builds.shiki.hu/temp/qbittorrent_3.3.0alpha_based_on_a2c48646ae20_rev2.7z

Do Chinese(people) expect Traditional by default?

Chocobo1 commented:
Yes! Now the legal text is correct.

>Do Chinese(people) expect Traditional by default?

This depends where he lives. People in Mainland China expects simplified Chinese, whereas people in Hong Kong & Taiwan expects Traditional.

sledgehammer999 commented:
So since, QLocale::script() isn't reliable we can't do anything else. We can choose one way or another. I am leaving the default as Traditional.
I should merge this, right?

Chocobo1 commented:
I can only guarantee detecting Traditional Chinese is working. And no other issue occurred when testing.
